src/osd/PG.cc

```diff
@@ -4049,6 +4049,7 @@ int PG::build_scrub_map_chunk(
   // objects
   vector<hobject_t> ls;
   vector<ghobject_t> rollback_obs;
+  osr->flush();
```

Comment:
The change is only needed because of the synchronous onreadable behavior change in BlueStore.  Before now, the OSD doesn't expect changes to be there until onreadable is called, which happens later when the creates/deletes are visible.